### PR TITLE
Fix: restore selection after transform block

### PR DESCRIPTION
**Pull request by lawvs**
- Restore selection after transform block

https://user-images.githubusercontent.com/18554747/209319944-db96fe6c-4350-4d10-84de-fe45e84dc6cc.mov

- Format bar will following current range always now.

- [x] Wait block update is too trick so we need a better solution.

Also mentioned at https://github.com/toeverything/blocksuite/pull/358

### CODE REVIEW

File: packages/blocks/src/page-block/utils/container-operations.ts [hunk 1]
```diff
@@ -73,22 +74,33 @@ export function deleteModels(page: Page, models: BaseBlockModel[]) {
...
+      // Replace selected block id
+      const blocks = selectedBlocks.filter(block => block.id === oldId);
+      // Because selectedBlocks maybe contains same block when only select one block, so we need to replace all of them
+      blocks.forEach(block => {
+        block.id = newId;
+      });
     }
   });
+  // XXX FIXME: wait for page update
+  await sleep(100);
```
Comment (himself65) [inline packages/blocks/src/page-block/utils/container-operations.ts]: Is there any signal for the page update?

File: packages/blocks/src/page-block/utils/container-operations.ts [hunk 2]
```diff
@@ -73,22 +74,33 @@ export function deleteModels(page: Page, models: BaseBlockModel[]) {
...
   page.captureSync();
-  modelsInRange.forEach(model => {
+  modelsInRange.forEach(async model => {
     assertFlavours(model, ['affine:paragraph', 'affine:list']);
     if (model.flavour === flavour) {
       page.updateBlock(model, { type });
     } else {
-      transformBlock(page, model, flavour, type);
+      const oldId = model.id;
+      const newId = transformBlock(page, model, flavour, type);
```
Comment (himself65) [inline packages/blocks/src/page-block/utils/container-operations.ts:120]: Can transform block function returns a promise that we don't need `sleep`

File: packages/blocks/src/__internal__/rich-text/rich-text.ts [hunk 3]
```diff
@@ -91,7 +91,8 @@ export class RichText extends LitElement {
     });
 
     page.attachRichText(model.id, this.quill);
-    page.awareness.updateLocalCursor();
+    // TODO it can cause cursor flickering, by need to evaluate impact
+    // page.awareness.updateLocalCursor();
```
Comment (lawvs) [inline packages/blocks/src/__internal__/rich-text/rich-text.ts:95]: After removing the `asyncFocusRichText` after the transform block, the `updateLocalCursor` will cause cursor flickering after transform.

Is the update here necessary? @doodlewind

**Review by doodlewind — APPROVED** [review]
LGTM

**Pull request merged**